packages/truffle-migrate/migration.js

```diff
@@ -97,50 +96,151 @@ class Migration {
       if (Migrations && Migrations.isDeployed()) {
         const message = `Saving migration to chain.`;
 
-        if (!this.dryRun){
+        if (!this.dryRun) {
           const data = { message: message };
-          await self.emitter.emit('startTransaction', data);
+          await self.emitter.emit("startTransaction", data);
         }
 
         const migrations = await Migrations.deployed();
         const receipt = await migrations.setCompleted(self.number);
 
-        if (!this.dryRun){
-          const data = {receipt: receipt, message: message };
-          await self.emitter.emit('endTransaction', data);
+        if (!this.dryRun) {
+          const data = { receipt: receipt, message: message };
+          await self.emitter.emit("endTransaction", data);
         }
       }
 
-      await self.emitter.emit('postMigrate', self.isLast);
+      await self.emitter.emit("postMigrate", self.isLast);
 
       // Save artifacts to local filesystem
       await options.artifactor.saveAll(resolver.contracts());
 
       deployer.finish();
 
       // Cleanup
-      if (self.isLast){
+      if (self.isLast) {
         self.emitter.clearListeners();
 
         // Exiting w provider-engine appears to be hopeless. This hack on
         // our fork just swallows errors from eth-block-tracking
         // as we unwind the handlers downstream from here.
-        if (self.options.provider && self.options.provider.engine){
+        if (self.options.provider && self.options.provider.engine) {
           self.options.provider.engine.silent = true;
         }
       }
       // Prevent errors thrown in the callback from triggering the below catch()
       process.nextTick(callback);
-    } catch(e) {
+    } catch (e) {
       const payload = {
-        type: 'migrateErr',
+        type: "migrateErr",
         error: e
       };
 
-      await self.emitter.emit('error', payload);
+      await self.emitter.emit("error", payload);
       deployer.finish();
       callback(e);
+    }
+  }
+
+  // ------------------------------------- Private -------------------------------------------------
+  /**
+   * Runs the legacy migration sequence by instantiating a
+   * legacy deployer using a legacy web3 provider.
```

Comment:
I don't think the last bit about legacy web3 provider is true anymore? (I'm assuming this code used to reference that `wrapLegacyProvider` function)